tool/tsh/db.go

```diff
@@ -156,7 +154,35 @@ func fetchDatabaseCreds(cf *CLIConf, tc *client.TeleportClient) error {
 	if trace.IsNotFound(err) {
 		return nil // No currently logged in profiles.
 	}
+	proxy, err := tc.ConnectToProxy(cf.Context)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	cluster, err := proxy.ConnectToCluster(cf.Context, tc.SiteName, true)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	defer cluster.Close()
+
 	for _, db := range profile.Databases {
+		mfaResp, err := cluster.IsMFARequired(cf.Context, &proto.IsMFARequiredRequest{
+			Target: &proto.IsMFARequiredRequest_Database{
+				Database: &proto.RouteToDatabase{
+					ServiceName: db.ServiceName,
+					Protocol:    db.Protocol,
+					Username:    cf.Username,
+					Database:    db.Database,
+				},
+			},
+		})
+		if err != nil {
+			return trace.Wrap(err)
+		}
+		if mfaResp.GetRequired() {
+			// Skip DB if databaseLogin command will require MFA.
+			// to prevent MFA prompt during tsh db ls command where DB certs are refreshed.
+			continue
+		}
```

Comment:
Have the same feeling, there is reason for  `tsh db ls`  to fetching  db credentials. I will remote it. 

>  This would simplify this code too then, right?

Refetching "active" DB credentials logic is also called during `tsh login` command, for DB where second factor is require it will prompt for MFA auth thus this code is needed to change this behavior.